### ISSUE TITLE
[DEFECT] Incorrect scoring if you encounter no defect throughout the run

### ISSUE THREAD

**Issue by ghost** [COMP3122-2324-Project/Simpsyber#1]
If you are lucky enough to encounter no defect throughout the run, you should have **0 Change Failure Rate** and **0 Mean Time to Recover**, which is perfect and deserves a straight A.

However, the scores seem to be deducted for no reason in this case:
<img width="1376" alt="" src="https://github.com/COMP3122-2324-Project/Simpsyber/assets/126453494/ead6a28f-c554-4ab8-8d05-1a6c93914232">

As the game design ties our performance to an overall score, that is, all we do in the game is to strive for a higher score, it is important to have the scoring system work as intended.

**Comment by ivanLeung-18** [comment]
I agree with your perspective. Normally, scores should not be deducted without a reason. I think you do not have encounter any event of Mean Time to Recover (MTTR) such as  recovering from any failure. For example, I try one times that I don't make any action on Kanban Page. the picture 1 can show the result, 'Mean Time to Recover' is displaying '--'. which mean  the initialization value. 

Picture 1:
![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/156775094/19a4c366-2013-4db6-9eb2-16feea48445f)

In my case, it shows numerical values like '0d' or '4.9d' since I have doing any recover or encounter the any defects (Picture 4) in the game. This could be the root cause of the unexpected score deductions.

Picture 2:

<img width="961" alt="螢幕截圖 2024-02-09 上午4 54 22" src="https://github.com/COMP3122-2324-Project/Simpsyber/assets/156775094/89a269d1-4807-437a-83d6-9085c5ce35c2">

Picture 3:
<img width="961" alt="螢幕截圖 2024-02-09 上午12 46 15" src="https://github.com/COMP3122-2324-Project/Simpsyber/assets/156775094/f6f7217e-1eea-4e4a-b235-ccb74c0eb2bf">

Picture 4:
<img width="70" alt="螢幕截圖 2024-02-09 上午4 56 49" src="https://github.com/COMP3122-2324-Project/Simpsyber/assets/156775094/4b0c9dfc-4449-423a-a8f3-bd1cf2e064e9">

**Comment by ghost** [comment]
@ivanLeung-18 From your results, I think the root cause can be the **Change Failure Rate**, since we do not have a number displayed in **Mean Time to Recover** unless we have **Change Failure Rate > 0**.

Also another issue here: how come **Change Failure Rate** is capped at B grade? Has anyone managed to achieve an A?

**Comment by ivanLeung-18** [comment]
@happytommy123  I agree your idea since if we don't make the any event of Change Failure Rate such as any defects exists. it would make the Change Failure Rate == 0. The Mean time to recover would be "--". there are demo video to show evidence. Video 1 is not trigger any event of Change Failure Rate. Video 2 trigger any event of Change Failure Rate such as any defects exists.

Video 1:

https://github.com/COMP3122-2324-Project/Simpsyber/assets/156775094/3f75c37b-cab7-453c-8791-b1dc40af7a57

Video 2:

https://github.com/COMP3122-2324-Project/Simpsyber/assets/156775094/41206a69-bfb2-4431-a292-cb3817b3b023
